core/src/main/java/com/linecorp/armeria/internal/server/DefaultServiceRequestContext.java

```diff
@@ -195,6 +198,9 @@ public DefaultServiceRequestContext(
         maxRequestLength = cfg.maxRequestLength();
         this.additionalResponseHeaders = additionalResponseHeaders;
         this.additionalResponseTrailers = additionalResponseTrailers;
+
+        hook(contextHook);
+        hook(this.cfg.contextHook());
```

Comment:
We don't need to set these hooks whenever a context is created.
The hook that is set to the service config is used if it's set.
If it isn't set the hook that is set to `ServerBuilder` is used instead.

You might want to refer to how a `SuccessFunction` is set to the service config to use the same logic for context hook. 😉 
https://github.com/line/armeria/blob/b604f3f31f4a571da7809e9d524d0b1576c840d4/core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java#L1242-L1247